hcl2template/types.required_plugins.go

```diff
@@ -99,19 +98,6 @@ func (cfg *PackerConfig) decodeImplicitRequiredPluginsBlock(k ComponentKind, blo
 	// this makes this simple.
 	componentName := block.Labels[0]
 
-	store := map[ComponentKind]packer.BasicStore{
-		Builder:       cfg.parser.PluginConfig.Builders,
-		PostProcessor: cfg.parser.PluginConfig.PostProcessors,
-		Provisioner:   cfg.parser.PluginConfig.Provisioners,
-		Datasource:    cfg.parser.PluginConfig.DataSources,
-	}[k]
-	if store.Has(componentName) {
```

Comment:
Good questions, it make be easier if I break out each set of changes into their own PR to help here. This PR contains a number of changes to make various Packer features work in Legacy mode. I described each use case in the description. 

For this particular use case you can see the details under **Update implicitly required plugins to work even in Legacy Mode**

Reply:
That said, I will add a test for the use case you just mentioned and see how things behave.

Reply:
haha - I wanted to tell you that you did a great job with the plugin requirements check. I was very curious about this as well. I found that it does now because the call ListInstallationsOptions does a good of detecting that a plugin has already been installed so no need to implicitly install. 

```
// Running init against template with required plugin block for github.com/hashicorp/amazon v1.0.1
~>  PACKER_LEGACY_MODE=on go run . init ../packer-templates/amazon-ebs/amazon-ebs_ubuntu_shell-local.pkr.hcl
Installed plugin github.com/hashicorp/amazon v1.0.1 in "~/.packer.d/plugins/github.com/hashicorp/amazon/packer-plugin-amazon_v1.0.1_x5.0_darwin_amd64"

// Running init against template with no required plugin block after a successful init for github.com/hashicorp/amazon v1.0.1
~>  PACKER_LEGACY_MODE=on go run . init ../packer-templates/amazon-ebs/amazon-ebs_ubuntu_shell-local.pkr.hcl
```

Reply:
nice !